File: Programming Basics with JS/08. More Exersices/cleverLilly.js
function solve (age, washingMachinePrice, toyPrice) {
    age = Number(age);
    washingMachinePrice = Number(washingMachinePrice);
    toyPrice = Number(toyPrice);
    let money = 10;
    let sumMoney = 0;
    let toys = 0;

    for(let i = 1; i<=age; i++){
        if(i % 2=== 0){
            sumMoney += money - 1;
            money += 10;
        } else {
            toys += 1
        }
    }

    sumMoney += toys * toyPrice;
    diff = sumMoney - washingMachinePrice;

    if(sumMoney>=washingMachinePrice){
        console.log(`Yes! ${diff.toFixed(2)}`);
    } else {
        diff = Math.abs(diff)
        console.log(`No! ${diff.toFixed(2)}`);
    }

}

solve("21", "1570.98", "3");

File: Programming Basics with Java/04. For Loop - Lab/src/CleverLilly.java
import java.util.Scanner;

public class CleverLilly {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int age = Integer.parseInt(scan.nextLine());
        double price = Double.parseDouble(scan.nextLine());
        int toyPrice = Integer.parseInt(scan.nextLine());
        int amountToys = 0;
        int money = 0;
        int giftMoney = 10;

        for (int i = 1; i<=age; i++){

            if(i % 2 == 0){
                money += giftMoney - 1;
                giftMoney += 10;
            } else {
                amountToys++;
            }
        }

        money += amountToys * toyPrice;


        if(money - price >= 0){
            System.out.printf("Yes! %.2f", Math.abs(money - price));
        } else {
            System.out.printf("No! %.2f", Math.abs(price - money));
        }


    }
}
